be/src/exprs/vectorized/like_predicate.cpp

```diff
@@ -190,6 +224,44 @@ ColumnPtr LikePredicate::regex_fn(FunctionContext* context, const Columns& colum
     return regex_match(context, columns, false);
 }
 
+ColumnPtr LikePredicate::like_fn_with_long_constant_pattern(FunctionContext* context, const Columns& columns) {
+    return match_fn_with_long_constant_pattern<true>(context, columns);
+}
+
+ColumnPtr LikePredicate::regex_fn_with_long_constant_pattern(FunctionContext* context, const Columns& columns) {
+    return match_fn_with_long_constant_pattern<false>(context, columns);
+}
+
+template <bool like>
+ColumnPtr LikePredicate::match_fn_with_long_constant_pattern(FunctionContext* context, const Columns& columns) {
+    auto state = reinterpret_cast<LikePredicateState*>(context->get_function_state(FunctionContext::THREAD_LOCAL));
+
+    const auto& value_column = VECTORIZED_FN_ARGS(0);
+    auto [all_const, num_rows] = ColumnHelper::num_packed_rows(columns);
+
+    ColumnViewer<TYPE_VARCHAR> value_viewer(value_column);
+    ColumnBuilder<TYPE_BOOLEAN> result(num_rows);
+
+    for (int row = 0; row < num_rows; ++row) {
+        if (value_viewer.is_null(row)) {
+            result.append_null();
+            continue;
+        }
+
+        bool v = false;
+        if constexpr (like) {
```

Comment:
there is a function whose name is like, you'd better to give this like another name.

Reply:
> there is a function whose name is like, you'd better to give this like another name.

Have fixed